lib/ansible/executor/task_executor.py

```diff
@@ -564,6 +564,9 @@ def _evaluate_failed_when_result(result):
                     if attempt < retries:
                         result['_ansible_retry'] = True
                         result['retries'] = retries
+                        # Add the original task name into the result, so TaskResult can consult it for its name
+                        result['task_name'] = wrap_var(self._task.get_name())
```

Comment:
It does, but for v2_runner_retry, it's a task before post_validate happens, so the result._task doesnt have the task name templated.